### PR TITLE
feat: Create lint for 'that that' -> 'that which'

### PR DESCRIPTION
**Pull request by grantlemons**
- **feat(that-which): add then_exact_word_or_lowercase to pattern matching**
- **feat(that-which): add lint for 'that that' -> 'that which'**

### CODE REVIEW

File: harper-core/src/linting/that_which.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    pattern: Box<dyn Pattern>,
+}
+
+impl Default for ThatWhich {
+    fn default() -> Self {
+        let mut pattern = WordPatternGroup::default();
+
+        pattern.add(
+            "that",
+            Box::new(
```
Comment (grantlemons) [inline harper-core/src/linting/that_which.rs:20]: Used twice -> refactor into own variable.

**Comment by grantlemons** [comment]
Needs some tests

File: harper-core/src/linting/that_which.rs (hunk 1 above)
Comment (elijah-potter) [reply harper-core/src/linting/that_which.rs:20]: You can solve this without using `Clone`. A smart-pointer for example.

**Review by elijah-potter — APPROVED** [review]
LGTM

**Pull request merged**